sphinx_gallery/gen_rst.py

```diff
@@ -93,6 +97,18 @@ def flush(self):
         self.file2.flush()
 
 
+class MyBytesIO(BytesIO):
+    """Helper to deal with unicode not having buffer interface on Py2.7"""
+    def write(self, data):
+        # "unicode" objects don't have a buffer interface on Py2.7,
+        # so we need to manually convert
+        if isinstance(data, unicode):
+            data = data.encode('utf-8')
+        super(MyBytesIO, self).write(data)
+
+    def getvalue(self):
+        return super(MyBytesIO, self).getvalue().decode('utf-8')
```

Comment:
This class is actually a bit weird since it derives from BytesIO but `.getvalue` returns a string.

It seems like this is working for me:

```python

class NonUnicodeFriendlyStringIO(StringIO):
    def write(self, data):
        if not isinstance(data, unicode):
            data = data.decode('utf-8')
        StringIO.write(self, data)
```

and then later:

```python
my_buffer = NonUnicodeFriendlyStringIO()
```